### ISSUE TITLE
LaTeX: Avoid section names at the end of a page

### ISSUE THREAD

**Issue by sebhub** [sphinx-doc/sphinx#6618]
I have a large document with lots of small sections. Currently, the likelihood that a section name is the last content on a page is quite high. This looks ugly. One way to prevent this is the following patch:
```patch
diff --git a/sphinx/texinputs/sphinx.sty b/sphinx/texinputs/sphinx.sty
index 2eebbb914..9198d739b 100644
--- a/sphinx/texinputs/sphinx.sty
+++ b/sphinx/texinputs/sphinx.sty
@@ -40,7 +40,7 @@
 % for \text macro and \iffirstchoice@ conditional even if amsmath not loaded
 \RequirePackage{amstext}
 \RequirePackage{textcomp}% "warn" option issued from template
-\RequirePackage{titlesec}
+\RequirePackage[nobottomtitles*]{titlesec}
 \@ifpackagelater{titlesec}{2016/03/15}%
  {\@ifpackagelater{titlesec}{2016/03/21}%
   {}%
```

**Comment by jfbu** [comment]
I am interested in some concrete source exhibiting the problem in latexpdf output. I tried such source

```rest
===============================
Welcome to FOO's documentation!
===============================

===
Foo
===

.. raw:: latex

   \pdfsetrandomseed 12345678

Section
=======

A

.. raw:: latex

   \vspace{\pdfuniformdeviate 10 mm}


B

```

with the section repeated 100 times to create some random spacing. But the circa 20 pages output has not a single isolated section title at a bottom of a page.

However maybe your use case has some lists or images after section titles?

**Comment by jfbu** [comment]
For a stronger test use `\pdfuniformdeviate 7 cm` rather. (for all 100 sections).

The point is that LaTeX has a mechanism for not allowing easily a section title to be left alone at bottom of page if followed by regular text. I am thus interested in concrete example (a screen shot at least).

**Comment by sebhub** [comment]
Yes, I was a bit surprised to see this in a LaTeX output. I created the attached example with `sphinx-quickstart`. It seems to be related to the `.. topic::` directive.
[index.txt](https://github.com/sphinx-doc/sphinx/files/3453322/index.txt)
[test.pdf](https://github.com/sphinx-doc/sphinx/files/3453327/test.pdf)

**Comment by jfbu** [comment]
Thanks, indeed the topic directive uses in latex rendering a framed box which encourages a pagebreak before it even if the element is a section title.

Your workaround can be implemented in conf.py of your project:

```
latex_elements = {
    'passoptionstopackages': r'\PassOptionsToPackage{nobottomtitles*}{titlesec}',
}
```

We need to examine if this option which works fine in this case does not have some other effect or incompatibility. Thus, and for little time available for investigation in coming weeks I am tagging it for examination for some unspecified future version. Any informed additional help on this will of course be welcome.

Thanks for reporting.

**Comment by jfbu** [comment]
PS: if you have time please submit a PR with your patch.

**Comment by tapaswenipathak** [comment]
Hello folks: I would be happy PRing, @sebhub, lemme know.

**Comment by jfbu** [comment]
Fixed at 53bbc27f1fd3c9432fa6f135c94334c1fd409222. Thanks for reporting and PR #6633!
